### DISCUSSION TITLE
the 3d box of the car is not`t very good

### DISCUSSION THREAD

**Discussion by tust13018211** [SysCV/qd-3dt#11] [Q&A]
Thanks for your work, I use my data to test the code. But I find the 3d box is non`t very good, I want to konw the reason?
![GIF](https://user-images.githubusercontent.com/83019441/118615794-abab5380-b7f3-11eb-9919-2b7139b8dcaf.gif)

**Comment by tust13018211** [comment]
the model I used is kitti，not trained by myself，maybe this cause the bad effect

**Comment by eborboihuc** [comment]
Thanks for trying our method and applying our code to your data.

You may want to take a look at the input data, especially on the camera matrix.
From the visualization, the camera you used might not share the same camera intrinsic with KITTI. Also, it seems the camera extrinsic is not updating or not reflecting your ego-motion.

**Comment by tust13018211** [comment]
I really appreciate your rapid reply! Firtstly, the camera intrinsic and camera extrinsic has been adapted to our realstic sence; Secondly, would you please tell me that if the camera extrinsic is updated in real time? 
As you can see, the trace of the car seems correct, so i think the camera extrinsic is right, but the 3d box is notn`t very good. Thank you again and expected for your reply.


![GIF](https://user-images.githubusercontent.com/83019441/118624613-e5805800-b7fb-11eb-8232-a8cdb05f3913.gif)
![GIF2](https://user-images.githubusercontent.com/83019441/118624992-32fcc500-b7fc-11eb-9c1c-78ea4224fed1.gif)

**Comment by tust13018211** [comment]
I have a another quesion about the camera extrinsic, why in the kitti data, the global position didn`t   multiply camera extrinsic, only the roattion multiply camera extrinsic. but in waymo and nuSceness data, all the global pose and rotation  multiply  camera extrinsic？ I have a little confuse about this

**Comment by tobiasfshr** [comment]
Hi,

thanks for trying out our work! In the top video, the cars are moving at a very similar speed like the ego car, therefore the visualization doesn't really say much about the extrinsics (note that the BEV visualization shows the scene in the current camera frame, not in world frame!). 
A more reliable way to check the extrinsics would be to take a sequence where the ego car passes a parked car, and see if the position of the parked car stays constant in world frame. The camera position will be updated at each input frame if that's your question regarding real-time update.
Your second question seems confusing, the camera extrinsic *is* the global position and rotation. It relates your current camera coordinate frame to some global ('world') frame. Finally, i'd try to use the nuScenes model, since this one was trained with much more real-world data and thus may generalize better to your scenario. The best would probably still be to fine-tune it on some labelled data from your camera setup.

Best,
Tobias

**Reply by tust13018211** [reply]
ok, thaks your reply, I will use the nuScenes model and will update the result

**Reply by BaophanN** [reply]
Did you successfully run and visualize the nuScenes model?

**Comment by Leonngm** [comment]
Hi,
i also want to test the model on my own data. May i ask you how you got the model running with your own data? Will it be possible to run the model on a pc with only cpu?
Thanks in advance!

Best, 
Leon

**Reply by tust13018211** [reply]
Hi, I think it must use the GPU, and the model you can downloat in the https://github.com/SysCV/qd-3dt/blob/main/readme/MODEL_ZOO.md

**Reply by Leonngm** [reply]
Thanks for your answer! Did you just structured your own data the same way it is in the KITTI dataset?

**Reply by tust13018211** [reply]
yes，I structured my own data as the kittti dataset, but I didn`t get a good result. Maybe I need to train my own data

**Reply by Leonngm** [reply]
Ok, got the data now. Which script did you run to get the video above with the 3D boxes?

**Reply by tust13018211** [reply]
you can run the code as the author wirte in the readme

**Reply by xhangHU** [reply]
I did not find a method for generate video in the readme. Can you tell me exactly how to do it？

**Reply by tust13018211** [reply]
you can set the "show" as ture，then you can get the video. parser.add_argument('--show', action='store_true', help='show results',default='True')  in the test_eval_video_exp

**Reply by BaophanN** [reply]
@tust13018211 Can you please give me the visualization code you have run to get the video. I do not understand why I follow exactly as the instruction. However, the qd3dt/core folder cannot be imported while other folders of qd3dt runs normally
